### PR TITLE
Convert docs introduction to mdx

### PR DESCRIPTION
**Pull request by Yofamb**
## Description

Make the Introduction page of the docs an MDX file.

## Related Issues

Addresses #17989 

## before
screencapture-gatsbyjs-org-docs-2019-10-14-09_29_06

## after
screencapture-localhost-5500-docs-2019-10-14-09_28_30

### CODE REVIEW

File: www/src/pages/docs/index.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+# Gatsby.js Documentation
+
+Gatsby is a blazing fast modern site generator for React.
+
+## Get Started
+
+There are two main ways to get started with Gatsby:
+
+1. <Link to="/tutorial/">Tutorial</Link>: Step-by-step instructions on how to install Gatsby and start a project: written for people without Gatsby or web development experience, though these learning resources have helped developers of all skill levels.
```
Comment (tesseralis) [inline www/src/pages/docs/index.mdx]: You can just do Markdown links:

```
[Tutorial](/tutorial)
```

File: www/src/pages/docs/index.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { Link } from "gatsby"
```
Comment (tesseralis) [inline www/src/pages/docs/index.mdx]: This file should be `/docs/docs/index.md`.

File: www/src/pages/docs/index.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+   - <Link to="/docs/gatsby-internals/">Gatsby Internals</Link>: Dig into how Gatsby works behind the scenes.
+   - <Link to="/docs/using-gatsby-professionally/"> Using Gatsby Professionally</Link>: Learn tips and tricks for how to explain Gatsby to others at work, so that you have more opportunities to work with Gatsby professionally.
+
+3. Check out the <Link to="/ecosystem/">Ecosystem</Link> libraries for Gatsby starters and plugins, as well as external community resources.
+
+## Start contributing
+
+Visit the <Link to="/contributing">Contributing</Link> section to find guides on the Gatsby community, code of conduct, and how to get started contributing to Gatsby.
+
+export default ({ children, location }) => (
```
Comment (tesseralis) [inline www/src/pages/docs/index.mdx]: You shouldn't need to do all this: the only thing component you'd need is `<EmailCaptureForm>`. Look at `template-doc-markdown` -- the content here is mostly the same as that, except for the email submission form.

Comment (Yofamb) [reply www/src/pages/docs/index.mdx]: Cheers, i'll have a look over the weekend. Thanks for the feedback

Comment (Yofamb) [reply www/src/pages/docs/index.mdx]: Hi @tesseralis I've fixed up the pull request. I still wasn't sure the best way to set up the email capture form. What's your thoughts on how i've achieved this?

File: www/src/templates/template-docs-markdown.js [hunk 4]
```diff
@@ -169,6 +173,7 @@ export const pageQuery = graphql`
         title
         overview
         issue
+        showSignupForm
```
Comment (Yofamb) [inline www/src/templates/template-docs-markdown.js]: potentially this should be called `enabledSignupForm` ?

File: www/src/templates/template-docs-markdown.js [hunk 5]
```diff
@@ -136,6 +137,9 @@ function DocsTemplate({ data, location, pageContext: { next, prev } }) {
                     See the issue relating to this stub on GitHub
                   </a>
                 )}
+                {page.frontmatter.showSignupForm && (
```
Comment (tesseralis) [inline www/src/templates/template-docs-markdown.js]: You should be able to just import and use the <EmailCaptureFrom> in [MDX](https://www.gatsbyjs.org/docs/mdx/writing-pages/).

In `index.md`:

```md
import EmailCaptureForm from 'www/src/components/EmailCaptureForm'

...content...

<EmailCaptureForm signupMessage="..."/>
```

**Comment by Yofamb** [comment]
@tesseralis I've updated everything now. Not sure why I made everything hard for myself at the start! this was a really simple issue. 

I am getting one test fail but i'm unsure why this is happening - could you please take a look and let me know what may be wrong? (it seems like un unrelated failure?). Once you've given the thumbs up ill mark this PR as ready to review.

**Comment by tesseralis** [comment]
@yofamb thanks! Next, could you verify that the page look the same before and after the change? Build the website locally and post a screenshot of the page to make sure it works.

**Comment by Yofamb** [comment]
Hi @tesseralis I've rebased the pull request, added screenshots and have compared the content from live docs and my branch. 

The only thing that i've noticed is different is what gets added to the title tag. See attached image Could you please review when you have the time. Cheers!

Screen Shot 2019-10-16 at 10 01 52 am

**Comment by tesseralis** [comment]
@yofamb this looks great! I'll ping in the rest of the gatsby team to see what to do about the title tag.

File: docs/docs/index.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+---
+title: Gatsby.js Documentation
```
Comment (tesseralis) [inline docs/docs/index.md]: You need to add `disableTableOfContents: true` to disable the right sidebar.

**Comment by marcysutton** [comment]
An update for the HTML title: I think it's fine to merge as-is. We can always update it later.

**Pull request merged**